envconfig.go

```diff
@@ -74,8 +78,23 @@ func Process(prefix string, spec interface{}) error {
 			f = f.Elem()
 		}
 
-		alt := ftype.Tag.Get("envconfig")
+		// Default to the field name as the env var name (will be upcased)
 		fieldName := ftype.Name
+
+		// Best effort to un-pick camel casing as separate words
+		if ftype.Tag.Get("multi_word") == "true" {
+			words := expr.FindAllStringSubmatch(ftype.Name, -1)
+			if len(words) > 0 {
+				var name []string
+				for _, words := range words {
+					name = append(name, strings.ToUpper(words[0]))
+				}
+
+				fieldName = strings.Join(name, "_")
```

Comment:
it's going to be more complicated to get right with the various cases below, but I think this behavior should be overriding the `key` variable instead of `fieldName` (the one time it makes a difference is when `fieldName` is provided separately from `key` in the `ParseError` wrapping).